enterprise/backend/src/metabase_enterprise/data_editing/params.clj

```diff
@@ -0,0 +1,207 @@
+                 :param-type  param-type})))))
+
+(defn- saved-param-input-type
+  [param-type viz-field]
+  (cond
+    ;; we could distinguish between inline-select and dropdown (which are both options for model action params)
+    (seq (:valueOptions viz-field))
+    "dropdown"
+
+    (= "text" (:inputType viz-field))
+    "textarea"
+
+    :else
+    (condp #(isa? %2 %1) (saved-param-base-type param-type viz-field)
+      :type/Date     "date"
+      :type/DateTime "datetime"
+      "text")))
+
+(defn- describe-saved-action
+  [& {:keys [action-id
+             param-mapping
+             row-delay]}]
+  (let [action              (-> (actions/select-action :id action-id
+                                                       :archived false
+                                                       {:where [:not [:= nil :model_id]]})
+                                api/read-check
+                                api/check-404)
+        param-id->viz-field (-> action :visualization_settings (:fields {}))
+        param-id->mapping   (u/index-by :parameterId param-mapping)]
+    {:title      (:name action)
+     :parameters (->> (for [param (:parameters action)
+                            ;; query type actions store most stuff in viz settings rather than the
+                            ;; parameter
+                            :let [viz-field     (param-id->viz-field (:id param))
+                                  param-mapping (param-id->mapping (:id param))]
+                            :when (and (not (:hidden viz-field))
+                                       (not= "hidden" (:visibility param-mapping)))]
+                        (u/remove-nils
+                         {:id            (:id param)
+                          :display_name  (or (:display-name param) (:name param))
+                          :param-mapping param-mapping
+                          :input_type    (saved-param-input-type (:type param) viz-field)
+                          :optional      (and (not (:required param)) (not (:required viz-field)))
+                          :nullable      true ; is there a way to know this?
+                          :readonly      (= "readonly" (:visibility param-mapping))
+                          :value         (param-value param-mapping row-delay)
+                          :value_options (:valueOptions viz-field)}))
+                      vec)}))
+
+(defn describe-unified-action
+  "Describe parameters of an unified action."
+  [unified scope input]
+  (let [scope (actions/hydrate-scope scope)]
+    ;; TODO: we didn't handle dashboard-action
+    (cond
+      ;; saved action
+      ;; hmm, having checked like this makes me insecure, what makes having an action-id enforces that this
+      ;; is an saved question? what if put an aciton-id on a row action for some reasons?
+      (:action-id unified)
+      (describe-saved-action :action-id (:action-id unified))
+
+      ;; table action
+      (:action-kw unified)
```

Comment:
I think it's worth calling out what a crazy assumption this is. It's actually gonna break on our Editable's built-in actions right now, but we can fix that in another PR.